src/logicblocks/event/processing/broker/strategies/singleton/factories.py

```diff
@@ -20,41 +17,45 @@
 
 
 class InMemorySingletonEventBrokerBuilder(
-    SingletonEventBrokerBuilder[(InMemoryEventStorageAdapter,)]
+    SingletonEventBrokerBuilder[(EventStorageAdapter,)]
 ):
     def dependencies(
-        self, adapter: InMemoryEventStorageAdapter
+        self, adapter: EventStorageAdapter
     ) -> SingletonEventBrokerDependencies:
         return SingletonEventBrokerDependencies(
-            event_source_factory=InMemoryEventStoreEventSourceFactory(
-                adapter=adapter
-            )
+            event_source_factory=EventStoreEventSourceFactory(adapter=adapter)
         )
 
 
 class PostgresSingletonEventBrokerBuilder(
     SingletonEventBrokerBuilder[
-        (ConnectionSettings, AsyncConnectionPool[AsyncConnection])
+        (
+            ConnectionSettings,
+            AsyncConnectionPool[AsyncConnection],
+            EventStorageAdapter | None,
+        )
     ]
 ):
     def dependencies(
         self,
         connection_settings: ConnectionSettings,
         connection_pool: AsyncConnectionPool[AsyncConnection],
+        adapter: EventStorageAdapter | None = None,
     ) -> SingletonEventBrokerDependencies:
+        event_storage_adapter = adapter or PostgresEventStorageAdapter(
+            connection_source=connection_pool
+        )
         return SingletonEventBrokerDependencies(
-            event_source_factory=PostgresEventStoreEventSourceFactory(
-                adapter=PostgresEventStorageAdapter(
-                    connection_source=connection_pool
-                )
+            event_source_factory=EventStoreEventSourceFactory(
+                adapter=event_storage_adapter
             )
         )
 
 
 def make_in_memory_singleton_event_broker(
     node_id: str,
     settings: SingletonEventBrokerSettings,
-    adapter: InMemoryEventStorageAdapter,
+    adapter: EventStorageAdapter,
```

Comment:
I don't think this is correct, I think you have to split it out into two different attributes.